# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/wr/WebRenderImageHost.cpp
# Commit: ce28584e3869
# Full Hash: ce28584e386976fc6df815260cdfebd8e6ffbadd
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2023-12-19 05:06:09
# Regressor Bug: 1829026
# File Overlap Count: 1
# Description:
#   Bug 1829026 - Update CanvasTranslator to work with DrawTargetWebgl. r=aosmond
#   
#   This adds the necessary infrastructure for CanvasTranslator to allocate DrawTargetWebgl
#   instead of just allocating TextureData, and to use RemoteTextureMap to handle sending
#   the DrawTargetWebgl frames to the compositor.
# ==============================================================================

diff -r bb4029ea69bb -r ce28584e3869 gfx/layers/wr/WebRenderImageHost.cpp
--- a/gfx/layers/wr/WebRenderImageHost.cpp	Mon Dec 18 18:10:46 2023 +0000
+++ b/gfx/layers/wr/WebRenderImageHost.cpp	Mon Dec 18 18:10:46 2023 +0000
@@ -143,6 +143,7 @@
       // Clear when RemoteTextureOwner is different.
       mPendingRemoteTextureWrappers.clear();
       mWaitingReadyCallback = false;
+      mWaitForRemoteTextureOwner = true;
     }
   }
 
@@ -222,7 +223,8 @@
 
     std::function<void(const RemoteTextureInfo&)> function;
     RemoteTextureMap::Get()->GetRemoteTextureForDisplayList(
-        wrapper, std::move(function));
+        wrapper, std::move(function), mWaitForRemoteTextureOwner);
+    mWaitForRemoteTextureOwner = false;
   }
 
   if (!texture ||